packages/builder/src/schemas.ts

```diff
@@ -536,7 +591,34 @@ export const cloneSchema = z
               message: `Target value: ${val} must match package formats: "package:version" or "package:version@preset" or be an interpolated value`,
             })
           )
-          .describe('Name of the package to provision'),
+          .refine(
+            (val) => {
+              const match = val.match(packageRegex);
+              if (match) {
+                const nameSize = new Blob([match!.groups!.name]).size;
+
+                return nameSize <= 32;
+              } else {
+                return true;
+              }
+            },
+            (val) => ({ message: `Package reference "${val}" is too long. Package name exceeds 32 bytes` })
+          )
+          .refine(
+            (val) => {
+              const match = val.match(packageRegex);
+
+              if (match) {
+                const versionSize = new Blob([match!.groups!.version]).size;
+
+                return versionSize <= 32;
+              } else {
+                return true;
+              }
+            },
+            (val) => ({ message: `Package reference "${val}" is too long. Package version exceeds 32 bytes` })
```

Comment:
These functions should not be repeated, maybe they can be refactored as a single new fn in `PackageReference` that's called `validate` and returns `undefined` in case the given ref is correct, or an error string in case it found one.

Reply:
Then, that `PackageReference.validate` fn can be used by `PackageReference.isValid`, `PackageReference.constructor` and also the schema validators

Reply:
There's an issue when importing from package through the schema file, which causes the action to register incorrectly, which is why I omitted using it in the schema section alltogether.

Reply:
However I could refactor them into a single function that takes a value and checks the length to be less or equal to 32 bytes, so that we avoid repetition.